File: app-android/feature/history/src/main/java/com/aidatainsight/android/feature/history/application/usecase/DeleteHistoryUseCase.kt
package com.aidatainsight.android.feature.history.application.usecase

import com.aidatainsight.android.feature.history.application.model.DeleteHistoryOutput
import com.aidatainsight.android.feature.history.application.model.HistoryRecordGroup
import com.aidatainsight.android.feature.history.application.model.HistoryStateSnapshot
import com.aidatainsight.android.feature.history.domain.HistoryRepository

class DeleteHistoryUseCase(
    private val repository: HistoryRepository,
) {
    suspend operator fun invoke(
        historyId: Int,
        existingGroups: List<HistoryRecordGroup>,
    ): DeleteHistoryOutput {
        repository.deleteHistory(historyId)
        val groups = existingGroups.mapNotNull { group ->
            val records = group.records.filter { it.id != historyId }
            if (records.isEmpty()) null else group.copy(records = records)
        }
        return DeleteHistoryOutput(
            deletedHistoryId = historyId,
            state = HistoryStateSnapshot(page = null, groups = groups),
        )
    }
}

File: app-ios/Packages/module-ai/Sources/ModuleAI/Application/UseCases/History/DeleteHistoryUseCase.swift
//
//  DeleteHistoryUseCase.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/8.
//

import Foundation

struct DeleteHistoryUseCase {
    private let repository: HistoryRepository

    init(repository: HistoryRepository) {
        self.repository = repository
    }

    func execute(
        recordGroups: [HistoryRecordGroup],
        historyId: Int
    ) async throws -> DeleteHistoryOutput {
        try await repository.deleteHistory(historyId: historyId)

        let updatedGroups = recordGroups.compactMap { group -> HistoryRecordGroup? in
            let records = group.records.filter { $0.id != historyId }
            guard records.isEmpty == false else { return nil }
            return HistoryRecordGroup(kind: group.kind, records: records)
        }

        return DeleteHistoryOutput(
            historyId: historyId,
            state: HistoryStateSnapshot(
                pageModel: nil,
                recordGroups: updatedGroups
            )
        )
    }
}
